### ISSUE TITLE
Dropped Materialized View is blocking table mutation (alter)

### ISSUE THREAD

**Issue by lazywei** [ByConity/ByConity#1077]
### Briefly describe the bug

We noticed a few buggy behavior related to materialized view and table mutations. See repro for more details.

### How to Reproduce

**Case 1**

1. Create a base table, and then create an aggregate merge tree table via Materialized View
2. Drop the materialized view (i.e. the agg table), say `ed.base_table_agg_5m_mv`
3. Run ALTER TABLE to add column

Sometimes this operation goes thru sometimes it gives the following error
```
Code: 60, e.displayText() = DB::Exception: Received from 10.170.68.34:9000. DB::Exception: Table ed.`base_table_agg_5m_mv` (235e20f3-a17e-4ce5-8580-eee019ab2ae7) doesn't exist in ed, database engine type: Cnch SQLSTATE: 42P01. (version 21.8.7.1)
```

**Case 2**

1. Create a base table, and a MV. Similar to case 1
2. Run ALTER TABLE DROP COLUMN to drop some column on the base table

```
Code: 524, e.displayText() = DB::Exception: Trying to ALTER DROP column `ed.reason` which is referenced by materialized view ['base_table_agg_5m_mv'] SQLSTATE: HY000 (version 21.8.7.1)
```

With some debugging it seems the underlying system calls for `getDependentViewsByColumn` and it returns the dropped materialized view even though the actual table is already dropped.

### Version

Master Branch

**Comment by kevinthfang** [comment]
This is being fixed.

**Comment by lazywei** [comment]
Hmm we seem to still be able to reproduce this with following

Create table
```
CREATE TABLE cwc_test
(
    `body` String
)
ENGINE = CnchMergeTree
ORDER BY body
SETTINGS storage_policy = 'cnch_default_s3'

---------

CREATE TABLE cwc_test_target
(
    `ibody` String
)
ENGINE = CnchMergeTree
ORDER BY ibody
SETTINGS storage_policy = 'cnch_default_s3'
```

Then Create MV
```
CREATE MATERIALIZED VIEW cwc_test_mv TO cwc_test_target (ibody String) AS SELECT body FROM cwc_test
```

Then Drop MV
```
DROP VIEW cwc_test_mv
```

Then run alter
```
ALTER TABLE cwc_test ADD COLUMN new_col String
```

Then it would raise exception
```
Code: 60, e.displayText() = DB::Exception:  Table ed_mt.cwc_test_mv (fd0abb2e-e8c9-48bb-8704-43d767afd02b) doesn't exist in ed_mt, database engine type: Cnch SQLSTATE: 42P01 (version 21.8.7.1)
```

The place this exception happens is in 
```
DatabaseCatalog::getTableImpl
```

This is the trace
```
[byconity-server-1] 2024.02.28 19:08:53.329306 [ 5296 ] {e11d2b51-aaf1-4608-b1f9-f399369722f7} <Error> executeQuery: Code: 60, e.displayText() = DB::Exception: DatabaseCatalog::getTableImpl -- Table ed_mt.cwc_test_mv (fd0abb2e-e8c9-48bb-8704-43d767afd02b) doesn't exist in ed_mt, database engine type: Cnch SQLSTATE: 42P01 (version 21.8.7.1) (from [::1]:51142) (in query: ALTER TABLE ed_mt.cwc_test ADD COLUMN new_col String), Stack trace (when copying this message, always include the lines below):

0. Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x225d4112 in /opt/byconity/bin/clickhouse
1. DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool) @ 0xec54c00 in /opt/byconity/bin/clickhouse
2. DB::Exception::Exception<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&) @ 0x139d5830 in /opt/byconity/bin/clickhouse
3. void std::__1::__optional_storage_base<DB::Exception, false>::__construct<int const&, char const (&) [88], std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(int const&, char const (&) [88], std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&) @ 0x1c069d07 in /opt/byconity/bin/clickhouse
4. DB::DatabaseCatalog::getTableImpl(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, std::__1::optional<DB::Exception>*) const @ 0x1c05c2af in /opt/byconity/bin/clickhouse
5. DB::DatabaseCatalog::getTable(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>) const @ 0x1c06184a in /opt/byconity/bin/clickhouse
6. DB::IStorage::getDependentViewsByColumn(std::__1::shared_ptr<DB::Context const>) const @ 0x1cdc41f8 in /opt/byconity/bin/clickhouse
7. DB::StorageCnchMergeTree::checkAlterInCnchServer(DB::AlterCommands const&, std::__1::shared_ptr<DB::Context const>) const @ 0x1ce7e2c3 in /opt/byconity/bin/clickhouse
8. DB::StorageCnchMergeTree::checkAlterIsPossible(DB::AlterCommands const&, std::__1::shared_ptr<DB::Context const>) const @ 0x1ce8180f in /opt/byconity/bin/clickhouse
9. DB::InterpreterAlterQuery::executeToTable(DB::ASTAlterQuery const&) @ 0x1c4d9cf6 in /opt/byconity/bin/clickhouse
10. DB::InterpreterAlterQuery::execute() @ 0x1c4d75ee in /opt/byconity/bin/clickhouse
11. DB::executeQueryImpl(char const*, char const*, std::__1::shared_ptr<DB::IAST>, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum, bool, DB::ReadBuffer*) @ 0x1ca43448 in /opt/byconity/bin/clickhouse
12. DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum, bool) @ 0x1ca3ebd5 in /opt/byconity/bin/clickhouse
13. DB::TCPHandler::runImpl() @ 0x1d6cc821 in /opt/byconity/bin/clickhouse
14. DB::TCPHandler::run() @ 0x1d6db0dc in /opt/byconity/bin/clickhouse
15. Poco::Net::TCPServerConnection::start() @ 0x2254984c in /opt/byconity/bin/clickhouse
16. Poco::Net::TCPServerDispatcher::run() @ 0x22549d2c in /opt/byconity/bin/clickhouse
17. Poco::PooledThread::run() @ 0x22630e3a in /opt/byconity/bin/clickhouse
18. Poco::ThreadImpl::runnableEntry(void*) @ 0x2262e9ec in /opt/byconity/bin/clickhouse
19. start_thread @ 0x7ea7 in /lib/x86_64-linux-gnu/libpthread-2.31.so
20. clone @ 0xfca2f in /lib/x86_64-linux-gnu/libc-2.31.so


0 rows in set. Elapsed: 0.044 sec. 

Received exception from server (version 21.8.7):
Code: 60. DB::Exception: Received from localhost:9000. DB::Exception: DatabaseCatalog::getTableImpl -- Table ed_mt.cwc_test_mv (fd0abb2e-e8c9-48bb-8704-43d767afd02b) doesn't exist in ed_mt, database engine type: Cnch SQLSTATE: 42P01. 
```

**Comment by luffyhwl** [comment]
@lazywei 
The merging process is still ongoing, and it is expected to be completed by next Wednesday for internal integration.

**Comment by lazywei** [comment]
@luffyhwl I see, thanks for the clarification. Would you mind pinging here once the fix is merged? Really appreciate it. Thanks

**Comment by kevinthfang** [comment]
This fix will be included in ByConity 0.4.0
